modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/MultiEncodeWorkflowOperationHandler.java

```diff
@@ -0,0 +1,545 @@
+    String sourceOption = StringUtils.trimToNull(operation.getConfiguration(name));
+    String[] options = (sourceOption != null) ? sourceOption.split(SEPARATOR) : null;
+    return (options);
+  }
+
+  /*
+   * Encode multiple tracks in a mediaPackage concurrently with different encoding profiles for each track. The encoding
+   * profiles are specified by names in a list and are the names used to tag each corresponding output. Each source
+   * track will start one operation on one worker. concurrency is achieved by running on different workers
+   *
+   * @param src The source media package
+   *
+   * @param operation the current workflow operation
+   *
+   * @return the operation result containing the updated media package
+   *
+   * @throws EncoderException if encoding fails
+   *
+   * @throws WorkflowOperationException if errors occur during processing
+   *
+   * @throws IOException if the workspace operations fail
+   *
+   * @throws NotFoundException if the workspace doesn't contain the requested file
+   */
+  private WorkflowOperationResult multiencode(MediaPackage src, WorkflowOperationInstance operation)
+          throws EncoderException, IOException, NotFoundException, MediaPackageException, WorkflowOperationException {
+    MediaPackage mediaPackage = (MediaPackage) src.clone();
+    // Check which tags have been configured
+    String[] sourceTags = getConfigAsArray(operation, "source-tags");
+    String[] sourceFlavors = getConfigAsArray(operation, "source-flavors");
+    String[] targetTags = getConfigAsArray(operation, "target-tags");
+    String[] targetFlavors = getConfigAsArray(operation, "target-flavors");
+    String tagWithProfileConfig = StringUtils.trimToNull(operation.getConfiguration("tag-with-profile"));
+    boolean tagWithProfile = tagWithProfileConfig != null && Boolean.parseBoolean(tagWithProfileConfig);
+
+    // Make sure either one of tags or flavors are provided
+    if (sourceFlavors == null && sourceTags == null) {
+      logger.info("No source tags or flavors have been specified, not matching anything");
+      return createResult(mediaPackage, Action.CONTINUE);
+    }
+    String[] profiles = getConfigAsArray(operation, "encoding-profiles");
+    if (profiles == null)
+      throw new WorkflowOperationException("Missing encoding profiles");
+
+    // Sort out the combinatorics of all the tags and flavors
+    List<ElementProfileTagFlavor> selectors = getSrcSelector(sourceFlavors, sourceTags, targetFlavors, targetTags,
+            profiles);
+
+    long totalTimeInQueue = 0;
+    Map<Job, JobInformation> encodingJobs = new HashMap<Job, JobInformation>();
+    // Find the encoding profiles - should only be one per flavor or tag
+    for (ElementProfileTagFlavor eptf : selectors) {
+      // Look for elements matching the tag and flavor
+      Collection<Track> elements = eptf.elementSelector.select(mediaPackage, true);
+      for (Track track : elements) { // For each source
```

Comment:
Maybe use a variable name like sourceTrack to avoid the need of comments describing the purpose of the variable